lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,1139 @@
+			'default'     => array(),
+			'description' => __( 'Array of column names to be searched.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'enum' => array( 'post_title', 'post_content', 'post_excerpt' ),
+				'type' => 'string',
+			),
+		);
+
+		$query_params['slug'] = array(
+			'description' => __( 'Limit result set to posts with one or more specific slugs.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'type' => 'string',
+			),
+		);
+
+		$query_params['status'] = array(
+			'default'           => 'publish',
+			'description'       => __( 'Limit result set to posts assigned one or more statuses.' ),
+			'type'              => 'array',
+			'items'             => array(
+				'enum' => array_merge( array_keys( get_post_stati() ), array( 'any' ) ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => array( $this, 'sanitize_post_statuses' ),
+		);
+
+		$query_params = $this->prepare_taxonomy_limit_schema( $query_params );
+
+		if ( 'post' === $this->post_type ) {
+			$query_params['sticky'] = array(
+				'description' => __( 'Limit result set to items that are sticky.' ),
+				'type'        => 'boolean',
+			);
+		}
+
+		/*
+		* I have temporarilly removed this because I believe it is incorrect,
+		* if I don't comment this out, then there are no results if
+		* the query loop block is not set to filter by format.
+		$query_params['format'] = array(
+			'default'     => 'standard',
+			'description' => __( 'Limit result set to posts assigned one or more formats.' ),
+			'type'        => 'array',
+			'items'       => array(
+				'anyOf' => array(
+					array(
+						'type' => 'string',
+						'enum' => array_values( get_post_format_slugs() ),
+					),
+					array(
+						'type'  => 'array',
+						'items' => array(
+							'type' => 'string',
+							'enum' => array_values( get_post_format_slugs() ),
+						),
+					),
+				),
+			),
+		);
+		*/
+
+		/**
+		 * Filters collection parameters for the posts controller.
+		 *
+		 * The dynamic part of the filter `$this->post_type` refers to the post
+		 * type slug for the controller.
+		 *
+		 * This filter registers the collection parameter, but does not map the
+		 * collection parameter to an internal WP_Query parameter. Use the
+		 * `rest_{$this->post_type}_query` filter to set WP_Query parameters.
+		 *
+		 * @since 4.7.0
+		 *
+		 * @param array        $query_params JSON Schema-formatted collection parameters.
+		 * @param WP_Post_Type $post_type    Post type object.
+		 */
+		return apply_filters( "rest_{$this->post_type}_collection_params", $query_params, $post_type );
+	}
+
+	/**
+	 * Prepares the 'tax_query' for a collection of posts.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param array           $args    WP_Query arguments.
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return array Updated query arguments.
+	 */
+	private function prepare_tax_query( array $args, WP_REST_Request $request ) {
```

Comment:
No changes were made here, the function is only included because it is private in the original class...